### DISCUSSION TITLE
可以用 radical pinyin 的词典，替换当前的 chaizi 词典

### DISCUSSION THREAD

**Discussion by mirtlecn** [Mintimate/oh-my-rime#29] [Ideas]
注意到已经对当前的 chaizi.dict.yaml 进行了修改，应该发现原先未经校对的词典的一些问题：

- 缺了大量常用字，和通用汉字规范表对比，缺了 2 - 3 百字
- 有不少格式错误（具体可以 diff 对比一下 radical.txt 和原先字表）
- 拆字错误：起笔撇画统一被注为了点画等
- pypinyin 自动注音大量注音错误：如丁的下部，为钩，注成了生僻字 jue；
- 不少未注音的笔画（当前词典大概有十几个部件没音，直接用的原字）

所以我花了一些时间校对了一份， https://github.com/mirtlecn/rime-radical-pinyin 你可以看看，有问题你也可以帮忙纠错。

**Comment by Mintimate** [comment]
确实比原本的`chaizi.dict`好很多；后续就换成这个啦。不过有个小疑问🤔，输入方案如果引入其他方案，可以覆写`speller`么？就比如在`radical-pinyin`的scheme使用的是全拼方案的情况下，双拼方案引入反查，似乎是无法通过覆写或者修改其他配置，实现激活`algebra_double_pinyin`配置片段。

**Comment by mirtlecn** [comment]
没明白你的问题，

雾凇这里有说双拼怎么修改 https://github.com/iDvel/rime-ice/issues/133 不晓得是不是回答了你的问题：

```txt
创建 radical_pinyin.custom.yaml 修改反查及辅码派生规则：
patch:
  # 修改为小鹤双拼的拼写派生规则，因为不在同一个文件了，前面要加上文件名
  speller/algebra:
    __include: radical_pinyin.schema.yaml:/algebra_flypy
```

**Reply by Mintimate** [reply]
这种书写方案的custom会把方案radical_pinyin的speller都修改。但是如果一个配置里既有双拼和全拼，都把radical_pinyin作为反查，都使用双拼或者全拼就显然不是想要的效果。

所以想问反查内有没有可以单独覆写speller的配置？我之前看rime方案介绍没有，雾凇也没有

**Reply by mirtlecn** [reply]
理解了，rime 没有这个选项。

大概只能复制（或者新建，然后 include 引入）另外一份 radical pinyin 的方案，改名为 radical double pinyin 之类，然后全拼指向 radical pinyin，双拼指向 radical double pinyin 这样。

rime 自带的朙月拼音就是这么搞的

**Comment by mirtlecn** [comment]
另外库里面还有一个辅码 lua，可以实现用拼音（或者其他方案）输入，用五笔、笔画、部首搜索候选，方便拼音选候选，减少重码的，你要觉得好用也可以试试：

比如，搜索第三声、第四声的 lu 字：

![2024年02月29日22时15分](https://github.com/Mintimate/oh-my-rime/assets/76689045/cdd5ec2d-5004-444a-a8b2-a513c7846b68)

搜索部件后面为 jin 的所有念 jin 的字词：

![2024年02月29日22时24分](https://github.com/Mintimate/oh-my-rime/assets/76689045/3793d9fb-63a8-4653-979a-61b8227a7b14)

在很多重码词里面找，笔画以 点横 开头的字词
![20240229222617](https://github.com/Mintimate/oh-my-rime/assets/76689045/7787a0cc-3dce-4361-ade4-fb8cce09b970)

雾凇里面有简单的用例，库里面也有文档。

**Reply by Mintimate** [reply]
这个有点难，我现在刚学会打小鹤的声码... 对于要辅码的操作，有点反应不过来😂

**Comment by mirtlecn** [comment]
@Mintimate 拉下词典更新，这两天把看到的生僻字全拆了。

**Reply by Mintimate** [reply]
感谢提醒，已经同步 💐
